### PR TITLE
PLANET-7154: Duplicate Feeds on form duplication

### PR DESCRIPTION
**Pull request by GP-Dan-Tovbein**
## Description

Ref: https://jira.greenpeace.org/browse/PLANET-7154

The current implementation comes from [PLANET-7053](https://jira.greenpeace.org/browse/PLANET-7053) and its [related PR](https://github.com/greenpeace/planet4-master-theme/pull/2006).

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [jupiter](https://www-dev.greenpeace.org/test-jupiter) | [admin](https://www-dev.greenpeace.org/test-jupiter/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-jupiter/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-jupiter) | [composer-local.json](https://github.com/greenpeace/planet4-test-jupiter/blob/main/composer-local.json)

:watch: 2023.05.15 12:32:09

**Review by Osong-Michael — APPROVED** [review]
👍 Looks good.

File: src/GravityFormsExtensions.php [hunk 1]
```diff
@@ -627,4 +628,26 @@ public function p4_gf_custom_confirmation_redirect($confirmation, $form)
 
         return $confirmation;
     }
+
```
Comment (sagarsdeshmukh) [inline src/GravityFormsExtensions.php:631]: Function `DocBlock ` is missing.

**Comment by sagarsdeshmukh** [comment]
Hi @GP-Dan-Tovbein , I tested this PR on my local and the feed was not duplicated when I duplicate the form. so I installed the analytics plugin on [test-jupiter](https://github.com/greenpeace/planet4-test-jupiter/blob/main/composer-local.json#L8) instance, but I am getting the same result. Did I am miss something? 🤔

**Comment by GP-Dan-Tovbein** [comment]
> Hi @GP-Dan-Tovbein , I tested this PR on my local and the feed was not duplicated when I duplicate the form. so I installed the analytics plugin on [test-jupiter](https://github.com/greenpeace/planet4-test-jupiter/blob/main/composer-local.json#L8) instance, but I am getting the same result. Did I am miss something? 🤔

@sagarsdeshmukh @Osong-Michael  I've fixed the issue. Looking at the below code 👇 
```php
$form_feed = array_filter(
            $feeds,
            function ($feed) use ($form_id) {
                if ($feed['form_id'] == $form_id) {
                    return $feed;
                }
            }
        );
```
The PHP Code Sniffer doesn't allow to use `==` for comparisons, you should use `===`. However, `$feed['form_id']` it's an `string` and `$form_id` it's an integer. Thus, `$feed['form_id'] == $form_id` is equals to `true` but `$feed['form_id'] === $form_id` is equals to `false`. Now I casted as an `integer` and it's working as expected.

For the previous tests, it was working properly because I didn't take care about its checks.

Please check it again!

**Review by sagarsdeshmukh — APPROVED** [review]
PR tested on local, Looks good to me 👍

**Pull request merged**